tests/test_graph.py

```diff
@@ -12,28 +12,34 @@
 name_sets = [['python2'],
              ['kivy']]
 bootstraps = [None,
-              Bootstrap.get_bootstrap('pygame', ctx),
               Bootstrap.get_bootstrap('sdl2', ctx)]
 valid_combinations = list(product(name_sets, bootstraps))
 valid_combinations.extend(
     [(['python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx)),
-     (['kivy', 'python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx))])
+     (['kivy', 'python3crystax'], Bootstrap.get_bootstrap('sdl2', ctx)),
+     (['flask'], Bootstrap.get_bootstrap('webview', ctx))])
 invalid_combinations = [[['python2', 'python3crystax'], None]]
 
 
 @pytest.mark.parametrize('names,bootstrap', valid_combinations)
 def test_valid_recipe_order_and_bootstrap(names, bootstrap):
-    get_recipe_order_and_bootstrap(ctx, names, bootstrap)
+    try:
+        get_recipe_order_and_bootstrap(ctx, names, bootstrap)
+    except Exception as e:
+        # pytest output is too terse, so do something better here:
+        print(
+            "test_valid_recipe_order_and_bootstrap FAILED " +
+            "with these specific parameters: " +
+            str((ctx, names, bootstrap))
+        )
+        raise e
 
 
 @pytest.mark.parametrize('names,bootstrap', invalid_combinations)
 def test_invalid_recipe_order_and_bootstrap(names, bootstrap):
     with pytest.raises(BuildInterruptingException) as e_info:
         get_recipe_order_and_bootstrap(ctx, names, bootstrap)
-    assert e_info.value.message == (
-        "Didn't find any valid dependency graphs. "
-        "This means that some of your requirements pull in conflicting dependencies."
-    )
+    assert e_info.value.message.lower().find("conflict") >= 0
```

Comment:
I understand you changed it so we don't have to update it too often whenever we change the wordings here, but well I think it's better to stay explicit or at least increase the context. Because just grepping `conflict` when it fails won't help that much and also reading the test feels a bit like "what's happening?"
And by the way why using `find("conflict")` over `"conflict" in` notation?

Reply:
> Because just grepping conflict when it fails won't help that much and also reading the test feels a bit like "what's happening"?

maybe I should just add a comment then?

Adding a fixed message is simply no longer possible, I could if at all grep for a specific sentence, but the function will no longer return a static text

>  And by the way why using find("conflict") over "conflict" in notation?

Because I'm silly :smile: I'll change it :+1:

Reply:
Yes a comment is also fine :+1: 
I'm missing where the message is no longer static, it looks pretty static to me or it's another message?
https://github.com/kivy/python-for-android/blob/a0ef2a09523ce74b877190ba4957e977234d962c/pythonforandroid/graph.py#L279

Reply:
@AndreMiras it calls `obvious_conflict_checker` first which produces a different, non-static message for all the obvious conflicts. so it is indeed no longer static

Reply:
should I still change something? did that clear up your questions?

Reply:
OK that's fine if it really bugs me, I'll do the change myself.
Basically I would have added a bit more context using the static part. e.g. `Conflict detected:` or `conflicting dependencies`. But it's fine as it is still way better than current master :+1: